### PR TITLE
Fix yarn file path resolutions \w missing manifest

### PR DESCRIPTION
**Pull request by feelepxyz**
Related PR: https://github.com/dependabot/dependabot-core/pull/1251

Fixes the case where the path dependency can't be found (only the folder
is actually needed to resolve the install).

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb [hunk 1]
```diff
@@ -164,27 +165,42 @@ def path_dependency_details(fetched_files)
...
           raise Dependabot::DependencyFileNotParseable, file.path
         end
 
-        dependency_objects.flat_map(&:to_a).
+        # Re-write the glob name to the targeted dependency name (which is used
+        # in the lockfile), for example "parent-pacakge/**/sub-dep/target-dep" >
+        # "target-dep"
+        resolution_deps = resolution_objects.flat_map(&:to_a).map do |name, v|
+          # Picking the last two parts that might include a scope
+          parts = name.split("/").reverse.first(2)
```
Comment (greysteil) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: This read a bit confusing to me. Maybe replace with:

```
parts = name.split("/").reverse.last(2)
parts.pop if parts.count == 2 && !parts.first.start_with?("@"))
[parts.join("/"), v]
```

Comment (greysteil) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: (FYI I updated the above comment to make it actually work.)

Comment (greysteil) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: Maybe also put this in a function (`convert_dependency_path_to_name(path)`) to appease rubocop?

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher/path_dependency_builder.rb [hunk 2]
```diff
@@ -118,13 +118,30 @@ def parsed_yarn_lock
           # to Yarn, which may not be safe
           @parsed_yarn_lock ||=
             begin
+              # Transform lockfile to parseable YAML by wrapping requirements in
+              # quotes, e.g. ("pkg@1.0.0":) and adding colon to nested
+              # properties (version: "1.0.0")
+              sanitize_requirement = lambda do |line|
```
Comment (greysteil) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher/path_dependency_builder.rb]: We should probably split this (ghetto Yarn.lock parsing) out into a separate class. If you've got loads on your plate I'm 👍 on keeping it here for now, though.

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb (hunk 1 above)
Comment (feelepxyz) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: Yes nice, cleaned this up and got rid of most of the disabled cops 😅

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher/path_dependency_builder.rb (hunk 2 above)
Comment (feelepxyz) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher/path_dependency_builder.rb]: Done!

**Comment by greysteil** [comment]
So much ❤️for splitting out the Yarn lockfile parser and adding tests for it!

**Pull request merged**